packages/data/addon/services/bard-metadata.js

```diff
@@ -74,16 +80,18 @@ export default Service.extend({
         )
         .then(payload => {
           //normalize payload
+          payload.source = options.dataSourceName;
           let metadata = get(this, '_serializer').normalize(payload);
 
           //set metadataLoaded property
           if (!(get(this, 'isDestroyed') || get(this, 'isDestroying'))) {
             //create metadata model objects and load into keg
-            this._loadMetadataForType('table', metadata.tables);
-            this._loadMetadataForType('dimension', metadata.dimensions);
-            this._loadMetadataForType('metric', metadata.metrics);
+            this._loadMetadataForType('table', metadata.tables, options.dataSourceName);
+            this._loadMetadataForType('dimension', metadata.dimensions, options.dataSourceName);
+            this._loadMetadataForType('metric', metadata.metrics, options.dataSourceName);
 
             this.set('metadataLoaded', true);
+            this.loadedDataSources.push(options.dataSourceName);
```

Comment:
Can we add a `namespace` option to the other accessor public methods (`all`, `getById`, etc)?

Reply:
Does all need it? I can add it to the others for sure.